src/sparse/impl/KokkosSparse_spmv_impl.hpp

```diff
@@ -318,12 +318,85 @@ spmv_beta_no_transpose (const KokkosKernels::Experimental::Controls& controls,
 			typename YVector::const_value_type& beta,
 			const YVector& y)
 {
-  typedef typename AMatrix::ordinal_type ordinal_type;
+  typedef typename AMatrix::non_const_ordinal_type ordinal_type;
   typedef typename AMatrix::execution_space execution_space;
 
   if (A.numRows () <= static_cast<ordinal_type> (0)) {
     return;
   }
+#if defined(KOKKOS_ENABLE_SERIAL) 
+  if(std::is_same<execution_space,Kokkos::Serial>::value) {
+    /// serial impl                                                                                         
+    typedef typename AMatrix::non_const_value_type value_type;
+    typedef typename AMatrix::non_const_size_type size_type;
+
+    const size_type *__restrict__ row_map_ptr = A.graph.row_map.data();
+    const ordinal_type *__restrict__ col_idx_ptr = A.graph.entries.data();
+    const value_type *__restrict__ values_ptr = A.values.data();
+
+    typename YVector::non_const_value_type *__restrict__  y_ptr = y.data();
+    typename XVector::const_value_type *__restrict__  x_ptr = x.data();
+
+    const typename YVector::non_const_value_type zero(0);
```

Comment:
I do not think there is anything wrong with this especially since it is on the host side, but in general using:
`const output_value_type zero = Kokkos::ArithTraits<output_value_type>::zero()`
can be more robust but that's no big deal.

Reply:
i think that this is just a style. ; )